Automattic/woocommerce-payments
# Add WooPay direct checkout data telemetry

Fixes https://github.com/Automattic/woopay/issues/2483

#### Changes proposed in this Pull Request

These changes are to start tracking the Proceed to Checkout button click event. This event also tracks whether WooPay direct checkout is being offered when the user clicks on the button.









#### Testing instructions

> [!IMPORTANT]  
> All cases should be tested against both classic and Blocks checkouts.

**Test: Proceed to Checkout tracking with direct checkout enabled**

1. Enable WooPay direct checkout: `npm run wp option update _wcpay_feature_woopay_direct_checkout 1`.
2. Go through each of the 4 scenarios in https://github.com/Automattic/woopay/pull/2390 test instructions.
3. After you click on the **Proceed to Checkout** button, there should be a request to `admin-ajax.php` and its payload should contain the event name `wcpay_proceed_to_checkout_button_click` and the property `woopay_direct_checkout` should evaluate to `true`.

**Test: Proceed to Checkout tracking with direct checkout disabled**

1. Disabled WooPay direct checkout: `npm run wp option update _wcpay_feature_woopay_direct_checkout 0`.
2. Go through each of the 4 scenarios in https://github.com/Automattic/woopay/pull/2390 test instructions.
3. After you click on the **Proceed to Checkout** button, there should be a request to `admin-ajax.php` and its payload should contain the event name `wcpay_proceed_to_checkout_button_click` and the property `woopay_direct_checkout` should evaluate to `false`.







-------------------


**Post merge**
